acme.go

```diff
@@ -116,7 +117,8 @@ type AuthzID struct {
 // Client implements ACME spec.
 type Client struct {
 	http.Client
-	Key *rsa.PrivateKey
+	// Key.Public() must be *rsa.PublicKey
+	Key crypto.Signer
```

Comment:
Nobody will see that comment and `Public().(*rsa.PublicKey)` will panic if `Public()` result is not of type `*rsa.PublicKey`.

It needs to be safer to use the `Client`. One way could be to replace `Key *rsa.PrivateKey` with 2 fields, `Signer` and `Pub`. Just an idea.